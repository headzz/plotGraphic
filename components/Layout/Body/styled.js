import styled from "styled-components";

export const BodyStyled = styled.div`
  width: 100%;
`;
