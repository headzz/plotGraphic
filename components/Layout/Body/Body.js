import PropTypes from "prop-types";
import React from "react";
import { BodyStyled } from "./styled";

/**
 * @component
 * @name Body
 * @description Defines the body for the layout.
 * @namespace components components/Body
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */
const Body = ({ children }) => {
  return <BodyStyled>{children}</BodyStyled>;
};

export default Body;

Body.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
