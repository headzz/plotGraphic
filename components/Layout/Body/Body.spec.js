import React from "react";
import { render, cleanup } from "@testing-library/react";
import Body from "./Body";

afterEach(cleanup);

test("render layout body", () => {
  const { container } = render(
    <Body>
      <div>1</div>
    </Body>
  );
  expect(container.firstChild).toHaveStyle("width: 100%");
  expect(container.firstChild).toMatchSnapshot();
});
