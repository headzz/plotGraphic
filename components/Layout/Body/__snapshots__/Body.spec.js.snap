// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render layout body 1`] = `
<div
  class="sc-bdVaJa UoSJW"
>
  <div>
    1
  </div>
</div>
`;
